doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,241 @@
+model.
+
+Gradient Boosting is a very powerful ensemble algorithm used for both 
+classification and regression tasks. It utilizes a series of weak learners 
+(specifically [decision tree regressors](decision_tree_regressor.md)) to arrive 
+closer and closer to a targeted label. Each 
+weak learner is trained on the error of the previous learner, thereby 
+reducing the error of the overall model.
+
+#### Simple usage example:
+
+```c++
+// Train a gradient boosting model on random numeric data and predict labels 
+// on test data: All data and labels are uniform random; 10 dimensional data, 
+// 5 classes.
+// Replace with a data::Load() call or similar for a real application.
+arma::mat dataset(10, 1000, arma::fill::randu); // 1000 points.
+arma::Row<size_t> labels =
+    arma::randi<arma::Row<size_t>>(1000, arma::distr_param(0, 4));
+arma::mat testDataset(10, 500, arma::fill::randu); // 500 test points.
+size_t numWeakLearners = 10;
+size_t numClasses = 5;
+
+mlpack::GradBoosting gb;                                // Step 1: create model.
+gb.Train(dataset, labels, numClasses, numWeakLearners); // Step 2: train model.
+arma::Row<size_t> predictions;
+gb.Classify(testDataset, predictions);                  // Step 3: classify points.
+
+// Print some information about the test predictions.
+std::cout << arma::accu(predictions == 2) << " test points classified as class "
+    << "2." << std::endl;
+```
+<p style="text-align: center; font-size: 85%"><a href="#simple-examples">More examples...</a></p>
+
+#### Quick links:
+
+ * [Constructors](#constructors): create `GradientBoosting` objects.
+ * [`Train()`](#training): train model.
+ * [`Classify()`](#classification): classify with a trained model.
+ * [Other functionality](#other-functionality) for loading, saving, and
+   inspecting.
+ * [Examples](#simple-examples) of simple usage and links to detailed example
+   projects.
+
+#### See also:
+
+ * [`AdaBoost`](adaboost.md)
+ * [`DecisionTree`](decision_tree.md)
+ * [`DecisionTreeRegressor`](decision_tree_regressor.md)
+ * [Random forests](random_forest.md)
+ * [mlpack classifiers](../../index.md#classification-algorithms)
+ * [Gradient Boosting on Wikipedia](https://en.wikipedia.org/wiki/Gradient_boosting)
+
+### Constructors
+
+ * `gb = GradBoosting()`
+   - Initialize model without training.
+   - You will need to call [`Train()`](#training) later to train the model before
+     calling [`Classify()`](#classification).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical and/or categorical data using default arguments for the
+     weak learners ([`DecisionTreeRegressor`s](decision_tree_regressor.md)).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
```

Comment:
Take a look at how the `DecisionTree` constructors are organized---the overloads that train on numerical data are organized in one place, and the overloads that do categorical data are in another.  Can you do the same here?  Right now only the categorical overloads are listed.

Also a bit of a side note, I would suggest making variable names shorter when possible.  This makes the rendered HTML display a bit better.  So, e.g., `info` instead of `datasetInfo`.  But, that is a minor comment, up to you if you want to handle it.